porter/v1/service.proto

```diff
@@ -60,6 +60,21 @@ message Service {
   // sleep is a flag to indicate that the service should be put to sleep
   // services that are asleep will have replicas set to 0 and autoscaling disabled
   optional bool sleep = 19;
+
+  // ComputeResources represents the compute resources that the service should run on.
+  ComputeResources compute_resources = 20;
+}
+
+// ComputeResources represents a list of compute resources. This allows us to distinguish between a nil list and an empty list.
+message ComputeResources {
+  // compute_resources is the list of compute resources
+  repeated ComputeResource compute_resources = 1;
+}
+
+// ComputeResource represents a node groups that the service should run on
```

Comment:
this comment should have something like "e.g. a node group" instead of strictly "represents a node group"